lib/qesapdeployment.pm

```diff
@@ -342,7 +342,19 @@ sub qesap_get_deployment_code {
     if (get_var('QESAP_INSTALL_VERSION')) {
         record_info('WARNING', 'QESAP_INSTALL_GITHUB_REPO will be ignored') if (get_var('QESAP_INSTALL_GITHUB_REPO'));
         record_info('WARNING', 'QESAP_INSTALL_GITHUB_BRANCH will be ignored') if (get_var('QESAP_INSTALL_GITHUB_BRANCH'));
-        my $ver_artifact = 'v' . get_var('QESAP_INSTALL_VERSION') . '.tar.gz';
+        my $ver_artifact;
+        if (get_var('QESAP_INSTALL_VERSION') eq 'latest') {
+            my $api_url = "https://api.github.com/repos/SUSE/qe-sap-deployment/releases/latest";
```

Comment:
I think this can cause some sporadic issues:
```
openqaworker15:~ # curl -s https://api.github.com/repos/SUSE/qe-sap-deployment/releases/latest
{"message":"API rate limit exceeded for 195.250.132.146. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)","documentation_url":"https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"}
```
I even restarted the VR, and was able to reproduce the error from openQA: https://openqaworker15.qa.suse.cz/tests/285988#step/configure/82

Now it seems to be working from ow15 again (some minutes after my failing runs), but without knowing details on the API limitations, I'd be hard pressed to use this in tests as it is.

Reply:
Changed it to avoid api usage
Nice catch!